src/packages/server/src/server.ts

```diff
@@ -0,0 +1,167 @@
+import {
+	SchemaBuilder,
+	graphweaverMetadata,
+	resolveAdminUiMetadata,
+	AdminUiMetadata,
+	fieldResolver,
+	enableFederation,
+	buildFederationSchema,
+	startTracing,
+	isTraceable,
+	GraphweaverPlugin,
+} from '@exogee/graphweaver';
+import { logger } from '@exogee/logger';
+import { ApolloServer, BaseContext, GraphQLRequest } from '@apollo/server';
+import { ApolloServerPluginInlineTrace } from '@apollo/server/plugin/inlineTrace';
+
+import { LogErrors, LogRequests, corsPlugin, dedupeGraphQL } from './apollo-plugins';
+import { StartServerOptions, startStandaloneServer, startServerless } from './integrations';
+import { GraphweaverConfig, mergeConfig } from './config';
+import { enableTracing } from './trace';
+import { pluginManager } from './plugins';
+import {
+	ExecuteOperationOptions,
+	GraphQLResponse,
+} from '@apollo/server/dist/esm/externalTypes/graphql';
+import { onRequestWrapper } from './integrations/utils';
+
+export default class Graphweaver<TContext extends BaseContext> {
+	server: ApolloServer<TContext>;
+	public schema: GraphQLSchema;
+	private graphweaverPlugins: Set<GraphweaverPlugin> = new Set();
+	private config: GraphweaverConfig = {
+		adminMetadata: { enabled: true },
+		apolloServerOptions: {
+			introspection: process.env.NODE_ENV !== 'production',
+		},
+		enableFederationTracing: false,
+		graphqlDeduplicator: {
+			enabled: true,
+		},
+	};
+
+	constructor(config?: GraphweaverConfig) {
+		logger.trace(`Graphweaver constructor called`);
+
+		// Assign default config
+		this.config = mergeConfig<GraphweaverConfig>(this.config, config ?? {});
+
+		startTracing({
+			instrumentations: this.config.openTelemetry?.instrumentations ?? [],
+			traceProvider: this.config.openTelemetry?.traceProvider,
+		});
+
+		// Configure the plugins for Graphweaver and Apollo
+		this.graphweaverPlugins = pluginManager.getPlugins();
+		const apolloPlugins = this.config.apolloServerOptions?.plugins || [];
+
+		for (const metadata of graphweaverMetadata.entities()) {
+			if (metadata.provider?.apolloPlugins && metadata.provider.apolloPlugins.length > 0) {
+				// only push unique plugins
+				const eMetadataProviderPlugins = metadata.provider.apolloPlugins.filter(
+					(plugin) => !apolloPlugins.includes(plugin)
+				);
+				apolloPlugins.push(...eMetadataProviderPlugins);
+			}
+		}
+
+		if (this.config.adminMetadata?.enabled) {
+			logger.trace(`Graphweaver adminMetadata is enabled`);
+			graphweaverMetadata.addQuery({
+				name: '_graphweaver',
+				description: 'Query used by the Admin UI to introspect the schema and metadata.',
+				getType: () => AdminUiMetadata,
+				resolver: resolveAdminUiMetadata(this.config.adminMetadata?.hooks),
+				directives: { inaccessible: true },
+			});
+		}
+
+		// Order is important here
+		const plugins = [
+			LogRequests,
+			LogErrors,
+			corsPlugin(this.config.corsOptions),
+			...apolloPlugins,
+			...(this.config.graphqlDeduplicator?.enabled ? [dedupeGraphQL] : []),
+		];
```

Comment:
Probably easier to read if we delete line 87 and just do:
```suggestion
		];
		
		if (this.config.graphqlDeduplicator?.enabled) plugins.push(dedupeGraphQL);
```

Reply:
This will be updated with https://exogee.atlassian.net/browse/EXOGW-348